test/unit/annotation_test.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+require 'test_helper'
+
+class AnnotationTest < ActiveSupport::TestCase
+  test 'not supported annotations are not allowed' do
+    subject = Annotation.new
+
+    subject.name = 'not_supported'
+
+    assert_not subject.valid?
+    assert subject.errors[:name].present?
+    assert subject.errors[:name].include? 'is not included in the list'
+  end
+
+  test ':name is mandatory' do
+    subject = Annotation.new
+
+    assert_not subject.valid?
+    assert subject.errors[:name].present?
+    assert subject.errors[:name].include? "can't be blank"
+  end
+
+  test 'a valid name is accepted' do
+    subject = Annotation.new
+
+    subject.name = 'managed'
```

Comment:
The tests with `['', ' ', nil]` suggest that annotations without a `value` are not to exist. Shouldn't we validate to false when `value` was not set?

Reply:
Done: https://github.com/3scale/porta/pull/3857/commits/71ebc6d6b8edbeef73d5f5c0a13f53e671d2a59c